File: python/socketd/socketd/transport/core/stream/SendStream.py
from socketd.transport.core.Costants import Constants
from socketd.transport.core.stream.StreamBase import StreamBase


class SendStream(StreamBase):

    def __init__(self, sid: str):
        super().__init__(sid, Constants.DEMANDS_ZERO, 0)

    def on_reply(self, reply):
        pass

    def is_done(self):
        return True

File: go/socketd/transport/stream/impl/send_stream.go
package impl

import (
	"socketd/transport/core/constant"
	"socketd/transport/core/message"
)

type SendStream struct {
	*StreamBase
}

func NewSendStream(sid string) *SendStream {
	return &SendStream{StreamBase: NewStreamBase(sid, constant.DEMANDS_ZERO, 0)}
}

// IsDone
/**
 * @Description 是否完成的
 */
func (s *SendStream) IsDone() bool {
	return true
}

func (s *SendStream) OnReply(reply *message.Frame) {

}
